src/nvim/api/buffer.c

```diff
@@ -411,10 +411,13 @@ void nvim_buf_set_lines(uint64_t channel_id,
 
 /// Gets a buffer-scoped (b:) variable.
 ///
-/// @param buffer     Buffer handle
-/// @param name       Variable name
-/// @param[out] err   Error details, if any
-/// @return Variable value
+/// @note Use buffer_get_changedtick() for “variable” `b:changedtick`.
+///       This method only gets variables from b: dictionary.
```

Comment:
This is one of the forward compatibility matters: we don’t allow functions starting from lowercase letter, commands starting with lowercase letter or non-letter, etc because if we do we will face the situation when adding new function or command will break user plugins. For the same reason various variables should not be special. It is OK to create new namespaced variables like `b:nvim[_.]changedtick`: one rule like “do not create new `.:nvim.*` variables” is easy to remember and comprehend.

`b:changedtick` as an existing interface is, of course, not the case, but making it work may be tricky. If you know how to set up things so that `copy(b:)['changedtick']` will get `b:changedtick` (*note*: `copy(b:)['changedtick']`, not yet another hack to make `nvim_buf_get_var` work as expected by special-casing `changedtick`: ideal solution should make it possible to retire all currently used changedtick-related hacks) it would be good. Otherwise it is better to deprecate `b:changedtick` and create a VimL function with an API counterpart.